### DISCUSSION TITLE
Issue with Snapshots

### DISCUSSION THREAD

**Discussion by TiddlyWiddly** [tsightler/ring-mqtt#335] [Support Q&A]
I think this is probably something I'm doing wrong. I've enabled snapshots and the camera in the config, but I don't see any new entities with "snapshot" in the entity name. Restarted the add-on as well as HA completely. My understanding from other posts is that it should create a new camera entity with "snapshot" that would update automatically and I could reference that in Lovelace.

```
enable_cameras: true
snapshot_mode: enabled
livestream_user: ''
livestream_pass: ''
enable_modes: false
enable_panic: false
beam_duration: 0
disarm_code: ''
mqtt_host: <auto_detect>
mqtt_port: <auto_detect>
mqtt_user: <auto_detect>
mqtt_password: <auto_detect>
branch: addon
debug: ring-*
ring_token: ''
location_ids:
  - ''
```

**Comment by tsightler** [comment]
Please see the documentation (or tell me if the documentation is incorrect) as "enabled" is not a valid value for snapshot_mode parameter.

**Reply by TiddlyWiddly** [reply]
Shit, sorry sir. I saw "disabled" and flipped to "enabled" intuitively without reading the documentation. Thank you for the gentle RTFM reminder.

**Reply by tsightler** [reply]
It's OK, you're not the first with this issue.  For 5.0 the parameter should go away as the snapshot feature will be set and enabled on a per-camera basis during runtime vs a static configuration for all devices, at least, that's the plan.

**Reply by blueflame66** [reply]
Hi, I could be another one who is not reading the instructions right.  I have the snapshot mode to 'interval' selected.   We are building a house and I would like to create some timelapse videos.  While testing with my doorbell on my current house (battery ring stick up with solar panel is on its way from Amazon) I am getting a snapshot every 30 seconds ...which is a bit overwhelming.   I am hoping I can set the battery camera to 5 minutes if possible (I understand the 10 min limitation on them and if I read your instructions right I might be able to go down to 5 minutes).   My question is this... besides setting snapshotmode to enable, is there any way on the environmental variables side to set the interval for the snapshots?  On my doorbell I have snapshots in the app set to 3 minutes but I am still getting one every 30 seconds....

**Reply by tsightler** [reply]
My guess is that you're getting 30 seconds by default due to the solar panel.  During initial startup ring-mqtt checks for the "operatingOnBattery" flag and, if it's true, tries to use the setting from the app or otherwise defaults the 600 seconds.  If the doorbell is actively powered by solar, then it's probably not flagged as operatingOnBattery, and this defaults to 30 seconds.  I don't own any battery powered Ring devices, so I'm guessing a little.

The snapshot interval can be changed individually for each camera, there is an MQTT topic as noted in the TOPIC.md file and you should also see an entity in HA (assuming you are using HA).  Unfortunately, in the 4.x branch, this value isn't persistent, so you have to reset the value any time you restart ring-mqtt, using an automation.  My hope is to implement device level persistence for these types of settings in the 5.0 release, but I haven't had much time to work on that implementation as of yet.  Still, it should be pretty easy to setup some type of automation that sets it any time it sees the 30 second value in state.
